minio/policy.py

```diff
@@ -528,36 +528,50 @@ def _get_permissions(s, resource, object_resource, matched_resource,
 
 
 # Returns policy of given bucket name, prefix in given statements.
-def get_policy(statements, bucket_name, prefix=''):
-    bucket_resource = _get_bucket_resource(bucket_name)[0]
-    object_resource = _get_object_resource(bucket_name, prefix)[0]
-
-    bucket_common_found = False
-    bucket_read_only = False
-    bucket_write_only = False
-    matched_resource = ''
-    obj_read_only = False
-    obj_write_only = False
-    for s in statements:
-        resources = _get_resource(s)
-        for resource in resources:
-            bucket_common_found, bucket_read_only, bucket_write_only, \
-                obj_read_only, obj_write_only = _get_permissions(
-                    s, resource, object_resource, matched_resource,
-                    bucket_resource, prefix, bucket_common_found,
-                    bucket_read_only, bucket_write_only)
-
-    policy = Policy.NONE
-    if bucket_common_found:
-        if (bucket_read_only and bucket_write_only and
-                obj_read_only and obj_write_only):
-            policy = Policy.READ_WRITE
-        elif bucket_read_only and obj_read_only:
-            policy = Policy.READ_ONLY
-        elif bucket_write_only and obj_write_only:
-            policy = Policy.WRITE_ONLY
-
-    return policy
+def get_policy(statements, bucket_name, prefix=""):
+    policies = list_policy(statements, bucket_name)
+    found_policy = policies.get(prefix)
+
+    if found_policy is None:
+        return Policy.NONE
+
+    return found_policy
+
+READ_WRITE_LENGTH = 5
+WRITE_ONLY_LENGTH = 4
+READ_ONLY_LENGTH = 1
+# Returns a list containing policies and their respective prefix name
+def list_policy(statements, bucket_name, prefix=""):
+    policies = {}
+    for statement in statements:
+        # get resource from statement
+        res = statement.get("Resource")[0]
+        res_name = res.strip(_AWS_RESOURCE_PREFIX).strip(bucket_name)
+
+        # if valid resource
+        if res_name != "":
+            res_name = res_name.replace("/", "", 1).replace("*", "", 1)
```

Comment:
I couldn't able to get removing `/` and `*`.  A resource may be like `"arn:aws:s3:::mybucket/hello*"`